### DISCUSSION TITLE
Constructor level properties

### DISCUSSION THREAD

**Discussion by YoussefSell** [dotnet/csharplang#6023] [Language Ideas] [answered]
with the introduction of Top-level statements & global usings, we now writing less unnecessary code, and I want to add on top of this idea a new feature request. it is not new, we already have it in TypeScript and it will be nice if we can add it to C# also.

### Current behavior

let's take for example the `Startup` class (even if it doesn't exist anymore), we inject `IConfiguration` into the constructor and initialize a prop.

```csharp
public partial class Startup
{
    public IConfiguration Configuration { get; }

    public Startup(IConfiguration configuration)
        => Configuration = configuration;
}
```
here we have 3 lines of code to capture the `IConfiguration` instance and save it to a prop, imagine a class with 5 constructor injections, how many lines will be!

### Suggested behavior
it will be nice if we can do the same as we do in TypeScript, So it looks like this:

```csharp
public partial class Startup
{
    public Startup(public IConfiguration Configuration);
}
```
so that we write it like that, and it could be lowered to the default prop initialization behavior.

I think this will be a good feature to implement, what do you think?

**Comment by HaloFour** [comment, marked as answer]
See:

https://github.com/dotnet/csharplang/issues/2691
https://github.com/dotnet/csharplang/issues/4024
https://github.com/dotnet/csharplang/discussions/4025
